src/metrics/rules/active_ruleset.go

```diff
@@ -216,18 +233,19 @@ func (as *activeRuleSet) mappingsForNonRollupID(
 			Tags:            snapshot.tags,
 			GraphitePrefix:  snapshot.graphitePrefix,
 		}
-		pipelines = append(pipelines, pipeline)
+		matchResult.forExistingID[idx].Pipelines = append(
+			matchResult.forExistingID[idx].Pipelines, pipeline,
+		)
 	}
+	matchResult.forExistingID[idx].CutoverNanos = cutoverNanos
 
 	// NB: The pipeline list should never be empty as the resulting pipelines are
 	// used to determine how the *existing* ID is aggregated and retained. If there
 	// are no rule match, the default pipeline list is used.
-	if len(pipelines) == 0 {
-		pipelines = metadata.DefaultPipelineMetadatas.Clone()
+	if len(matchResult.forExistingID[idx].Pipelines) == 0 {
+		matchResult.forExistingID[idx].Pipelines = metadata.DefaultPipelineMetadatas.Clone()
```

Comment:
we should avoid Cloning, it's just generating more heap allocations. It would make more sense to have a method like `PipelineMetadata.SetDefault()`